content/artifacts/artifacts-finder/plugin.js

```diff
@@ -120,36 +136,50 @@ class ArtifactsFinder {
   }
 
   prospectArtifacts() {
-    if (this.pendingPlanets.length >= MAX_PROSPECTING) {
+    let gearId = gear().id;
+    let gearPlanet = whereIsGear();
+    let currentBlockNumber = df.contractsAPI.ethConnection.blockNumber;
+
+    if (!gearPlanet) {
       return;
     }
-    let currentBlockNumber = df.contractsAPI.ethConnection.blockNumber;
-    let planets = Array.from(df.getMyPlanets()).filter(canHaveArtifact);
-    planets.forEach(planet => {
-      if (isFindable(planet, currentBlockNumber)) {
-        this.pendingPlanets.push(planet);
+    if (isProspectable(gearPlanet)) {
+      if (!gearPlanet.transactions?.hasTransaction(isUnconfirmedProspectPlanetTx)) {
+        df.prospectPlanet(gearPlanet.locationId);
```

Comment:
I hade done something similar in another plugin with await/async syntax:
```js
    const uctx = await df.prospectPlanet(planet.locationId)
    await uctx.confirmedPromise // wait until confirmed
    while (df.getPlanetWithId(planet.locationId).prospectedBlockNumber === undefined) {
      await sleep(blockTimeInMs)
    }
```
The code above guarantees will wait until the planet has finished prospecting.